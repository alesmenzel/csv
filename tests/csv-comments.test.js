const fs = require('fs')
const path = require('path')
const {pipeline} = require('stream')
const { promisify } = require('util')
const Parser = require('../csv-parser')
const Collect = require('./collect')

const pipelineAsync = promisify(pipeline)

it('CSV: relax comments', async () => {
  const parser = new Parser({ relaxComments: true })
  const collect = new Collect()
  const input = fs.createReadStream(path.resolve(__dirname, 'csv-comments.csv'))
  const comments = []
  parser.on('comment', (comment) => {
    comments.push(comment)
  })
  await pipelineAsync(input, parser, collect)
  expect(collect.getData()).toMatchSnapshot('csv-comments.csv')
  expect(comments).toMatchSnapshot('csv-comments.csv--comments')
})
