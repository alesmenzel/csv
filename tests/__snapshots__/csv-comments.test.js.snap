// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV: relax comments: csv-comments.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "b",
    "c",
  ],
  Array [
    "x",
    "y",
    "z",
  ],
]
`;

exports[`CSV: relax comments: csv-comments.csv--comments 1`] = `
Array [
  " comment",
  " this is a comment",
  " multiline",
  " next line",
  " and another one",
]
`;
